File: docs/snippets/all/archetypes/scalars_column_updates.py
"""
Update a scalar over time, in a single operation.

This is semantically equivalent to the `scalar_row_updates` example,
albeit much faster.
"""

from __future__ import annotations

import numpy as np

import dalaran as dl

dl.init("dalaran_example_scalar_column_updates", spawn=True)

times = np.arange(0, 64)
scalars = np.sin(times / 10.0)

dl.send_columns(
    "scalars",
    indexes=[dl.TimeColumn("step", sequence=times)],
    columns=dl.Scalars.columns(scalars=scalars),
)

File: docs/snippets/all/archetypes/scalars_column_updates.rs
//! Update a scalar over time, in a single operation.
//!
//! This is semantically equivalent to the `scalar_row_updates` example, albeit much faster.

use dalaran::TimeColumn;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new(
        "dalaran_example_scalar_column_updates",
    )
    .spawn()?;

    let times = TimeColumn::new_sequence("step", 0..64);
    let scalars = (0..64).map(|step| (step as f64 / 10.0).sin());

    rec.send_columns(
        "scalars",
        [times],
        dalaran::Scalars::new(scalars).columns_of_unit_batches()?,
    )?;

    Ok(())
}
